### PR TITLE
Phi fill diagonal

### PR DESCRIPTION
**Pull request by zhiboniu**
### PR types

Performance optimization
### PR changes

OPs
### Describe

transfer fill_diagonal kernels to Phi for new dygraph mode

### CODE REVIEW

File: paddle/phi/infermeta/unary.cc [hunk 1]
```diff
@@ -749,6 +749,13 @@ void ExpandInferMeta(const MetaTensor& x,
   }
 }
 
+void FillDiagonalInferMeta(
+    const MetaTensor& x, float value, int offset, bool wrap, MetaTensor* out) {
+  auto x_dims = x.dims();
+  out->set_dims(x_dims);
```
Comment (YuanRisheng) [inline paddle/phi/infermeta/unary.cc:799]: out做一下nullptr检查吧，nullptr需要抛出异常

File: paddle/phi/kernels/gpu/fill_diagonal_kernel.cu [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+                        float value,
+                        int offset,
+                        bool wrap,
+                        DenseTensor* out) {
+#ifdef __HIPCC__
+  const int64_t kMaxBlockDim = 256;
+#else
+  const int64_t kMaxBlockDim = 512;
+#endif
+  paddle::framework::TensorCopy(x, ctx.GetPlace(), out);
```
Comment (YuanRisheng) [inline paddle/phi/kernels/gpu/fill_diagonal_kernel.cu]: 使用phi下copy替换

File: python/paddle/fluid/tests/unittests/test_tensor_fill_diagonal_.py [hunk 3]
```diff
@@ -13,6 +13,7 @@
 # limitations under the License.
 
 import paddle.fluid as fluid
+from op_test import OpTest
```
Comment (YuanRisheng) [inline python/paddle/fluid/tests/unittests/test_tensor_fill_diagonal_.py]: 这个文件不需要改了吧

File: paddle/phi/infermeta/unary.cc (hunk 1 above)
Comment (zhiboniu) [reply paddle/phi/infermeta/unary.cc:799]: done

File: paddle/phi/kernels/gpu/fill_diagonal_kernel.cu (hunk 2 above)
Comment (zhiboniu) [reply paddle/phi/kernels/gpu/fill_diagonal_kernel.cu]: done

File: python/paddle/fluid/tests/unittests/test_tensor_fill_diagonal_.py (hunk 3 above)
Comment (zhiboniu) [reply python/paddle/fluid/tests/unittests/test_tensor_fill_diagonal_.py]: done

**Review by YuanRisheng — APPROVED** [review]

**Pull request merged**